internal/graph/check.go

```diff
@@ -152,6 +152,13 @@ func WithCachedResolver(opts ...CachedCheckResolverOpt) LocalCheckerOption {
 	}
 }
 
+func WithSingleflightResolver() LocalCheckerOption {
```

Comment:
```suggestion
func WithSingleFlightResolver() LocalCheckerOption {
```

Reply:
Naming is still a bit TBD but going by the package name, its a single word AFAICT.

Reply:
The package name is a single word due to Go's package name constraints, but the pattern is composed of 2 words, single flight :D fly only once! 🛩️

Reply:
The implementation of de-duplicated check resolution is always subject to change underneath the covers, but the purpose of the CheckResolver will always remain the same. For this reason I believe a more appropriate name would be `DedupeCheckResolver` or `DeduplicatedCheckResolver` etc.. If we swap from `singleflight` to some other implementation then the name still holds and the underlying implementation is of no matter.